src/core/context.rs

```diff
@@ -1,52 +1,65 @@
 use std::marker::PhantomData;
 use std::sync::Arc;
 
-use crate::core::types::{precomputed_square, RcVar, Type, Variable, VariableFactory};
 use crate::jet::Jet;
+use crate::types;
 
 /// Context for constructing a Simplicity program
 #[allow(dead_code)]
 pub struct Context<J: Jet> {
-    pub(crate) naming: VariableFactory,
-    pow2: Vec<RcVar>,
+    pub(crate) naming: types::variable::Factory,
+    pow22: Vec<types::Type>,
+    unit_ty: types::Type,
```

Comment:
Maybe add a comment that this is required so that the unit that the `pow2` types refer to is not dropped.

Reply:
It's not required, it's purely an efficiency thing so that we don't construct a gazillion `Unit` types and compute its TMR over and over.

Reply:
Ok, that makes sense. To avoid future confusion, let's add a comment